modules/check_in/app/services/v2/lorota/service.rb

```diff
@@ -90,6 +99,37 @@ def appointment_identifiers
           namespace: 'check-in-lorota-v2-cache'
         )
       end
+
+      private
+
+      def token_error_handler(e)
+        if e.original_status == 401
+          error_message_handler(e)
+        else
+          raise e
+        end
+      end
+
+      def error_message_handler(e)
+        case Oj.load(e.original_body).fetch('error')
+        when 'lastName does not match with current record',
+          'SSN4 does not match with current record',
+          'dob does not match with current record',
+          'lastName or dob does not match with current record'
+          retry_attempt_count = redis_client.retry_attempt_count(uuid: check_in.uuid) || 0
+          if retry_attempt_count < max_auth_retry_limit
+            redis_client.save_retry_attempt_count(uuid: check_in.uuid, retry_count: retry_attempt_count + 1)
+            raise e
+            # else
+            # call delete endpoint
+            # throw 410 exception
+          end
+        when 'UUID not found'
+          # raise 404 custom exception
```

Comment:
Should the 404 custom exception be implemented or is there a specific reason to exclude it for now?

Reply:
Currently, we don't get 404 from the dependency service.  Including custom exception code, would make PR further big.  Planning to add that in a separate PR.